zigbeegeneric/integrationpluginzigbeegeneric.cpp

```diff
@@ -287,32 +287,12 @@ void IntegrationPluginZigbeeGeneric::setupThing(ThingSetupInfo *info)
         }
 
         // Read initial attribute values
-        ZigbeeClusterReply *reply = thermostatCluster->readAttributes({ZigbeeClusterThermostat::AttributeLocalTemperature,
-                                                                       ZigbeeClusterThermostat::AttributeOccupiedHeatingSetpoint,
-                                                                       ZigbeeClusterThermostat::AttributePIHeatingDemand,
-                                                                       ZigbeeClusterThermostat::AttributePICoolingDemand});
-        connect(reply, &ZigbeeClusterReply::finished, thing, [=](){
-            if (reply->error() != ZigbeeClusterReply::ErrorNoError) {
-                qCWarning(dcZigbeeGeneric()) << "Reading thermostat attributes finished with error" << reply->error();
-                return;
-            }
-
-            QList<ZigbeeClusterLibrary::ReadAttributeStatusRecord> attributeStatusRecords = ZigbeeClusterLibrary::parseAttributeStatusRecords(reply->responseFrame().payload);
-            foreach (const ZigbeeClusterLibrary::ReadAttributeStatusRecord &record, attributeStatusRecords) {
-                if (record.attributeId == ZigbeeClusterThermostat::AttributeLocalTemperature) {
-                    thing->setStateValue(thermostatTemperatureStateTypeId, record.dataType.toInt16() * 0.01);
-                }
-                if (record.attributeId == ZigbeeClusterThermostat::AttributeOccupiedHeatingSetpoint) {
-                    thing->setStateValue(thermostatTargetTemperatureStateTypeId, record.dataType.toInt16() * 0.01);
-                }
-                if (record.attributeId == ZigbeeClusterThermostat::AttributePIHeatingDemand) {
-                    thing->setStateValue(thermostatHeatingOnStateTypeId, record.dataType.toUInt8() > 0);
-                }
-                if (record.attributeId == ZigbeeClusterThermostat::AttributePICoolingDemand) {
-                    thing->setStateValue(thermostatCoolingOnStateTypeId, record.dataType.toUInt8() > 0);
-                }
-            }
-        });
+        thermostatCluster->readAttributes({ZigbeeClusterThermostat::AttributeLocalTemperature,
+                                           ZigbeeClusterThermostat::AttributeOccupiedHeatingSetpoint,
+                                           ZigbeeClusterThermostat::AttributeMinHeatSetpointLimit,
+                                           ZigbeeClusterThermostat::AttributeMaxHeatSetpointLimit,
+                                           ZigbeeClusterThermostat::AttributePIHeatingDemand,
+                                           ZigbeeClusterThermostat::AttributePICoolingDemand});
```

Comment:
I remember there is  limit on how many attributes you can read with one request, maybe worth a check in the lib / spec how many are supported.